tests/AutoReview/ProjectCode/DocBlockParserTest.php

```diff
@@ -0,0 +1,117 @@
+<?php
+/**
+ * This code is licensed under the BSD 3-Clause License.
+ *
+ * Copyright (c) 2017, Maks Rafalko
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * * Redistributions of source code must retain the above copyright notice, this
+ *   list of conditions and the following disclaimer.
+ *
+ * * Redistributions in binary form must reproduce the above copyright notice,
+ *   this list of conditions and the following disclaimer in the documentation
+ *   and/or other materials provided with the distribution.
+ *
+ * * Neither the name of the copyright holder nor the names of its
+ *   contributors may be used to endorse or promote products derived from
+ *   this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+declare(strict_types=1);
+
+namespace Infection\Tests\AutoReview\ProjectCode;
+
+use Generator;
+use PHPUnit\Framework\TestCase;
+
+final class DocBlockParserTest extends TestCase
+{
+    /**
+     * @dataProvider docBlocksProvider
+     */
+    public function test_it_can_parse_PHP_doc_blocks(string $docBlock, string $expected): void
+    {
+        $actual = DocBlockParser::parse($docBlock);
+
+        $this->assertSame($expected, $actual);
+    }
+
+    public function docBlocksProvider(): Generator
+    {
+        yield ['', ''];
+
+        yield [
+            <<<'PHP'
+/**
+ * This is a
+ * multi-line
+ * doc-block
+ */
+PHP
+            ,
+            <<<'TEXT'
+This is a
+multi-line
+doc-block
+TEXT
+        ];
+
+        yield [
+            <<<'PHP'
+/**
+ * Single line doc-block
+ */
+PHP
+            ,
+            'Single line doc-block',
+        ];
+
+        yield [
+            <<<'PHP'
+   /**
+ * This is a
+ * multi-line  
+   * doc-block
+ * with weird indentation
+    */
+PHP
+            ,
+            <<<'TEXT'
+This is a
+multi-line
+doc-block
+with weird indentation
+TEXT
+        ];
+
+        yield [
+            <<<'PHP'
+/** Inlined doc-block */
+PHP
+            ,
+            'Inlined doc-block',
+        ];
+
+        yield [
+            <<<'PHP'
+// Comment
```

Comment:
Will these even be picked up by the parser?

This may be a weird result, but its not a use case the class needs to support anyways, i'd say remove the test here as well

Reply:
>Will these even be picked up by the parser?

Well the parser itself don't pick anything actually, it's the reflection. But to answer the question, it doesn't.

It's more to document the case and pin a behaviour than anything: i.e. later we can change the behaviour in this case and we know how it behaved before.